### PR TITLE
Migration script, makefile, readme, github actions

### PR DESCRIPTION
**Pull request by klatys**

### CODE REVIEW

**Review by mrlm-xyz — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+			Page:    1,
+			PerPage: 100,
+		},
+		OrderBy: gitlab.String("created_at"),
+		Sort:    gitlab.String("asc"),
+	}
+	for {
+		mergeRequests, response, _ := gitlabClient.MergeRequests.ListProjectMergeRequests(gitlabProject.ID, &gitlabMROptions)
+		for _, mr := range mergeRequests {
+			if mr.State == "closed" || mr.State == "merged" {
```
Comment (skooda) [inline main.go]: Could be flatter... (separate functions)

File: main.go [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+			threadArgs := git.CreateThreadArgs{
+				CommentThread: threadInit,
+				RepositoryId:  pullRequest.Repository.Name,
+				PullRequestId: pullRequest.PullRequestId,
+				Project:       pullRequest.Repository.Project.Name,
+			}
+			createdThread, err := azdoClient.CreateThread(azdoCtx, threadArgs)
+			if err != nil {
+				log.Error(err)
+				continue
```
Comment (skooda) [inline main.go]: Not sure what is content of err API response, but adding a ProjectName and other metadata to logged error would be great, for later investigation.

**Review by skooda — APPROVED** [review]
Minor suggestions.

File: main.go (hunk 1 above)
Comment (klatys) [reply main.go]: valid, thanks! Refactored

File: main.go (hunk 2 above)
Comment (klatys) [reply main.go]: valid, thanks! expanded the logging

**Review by mrlm-xyz — CHANGES_REQUESTED** [review]
Nothing against the implementation itself but I'm little bit worried about the error handling approach which is introducing a mixture of different approaches from explicitly ignored errors to mixing of error return as log.error and nil state to higher level with log.fatal which leads to process.exit and it is introducing different behavior on different execution unit levels. This should be serialized to unified approach.

Once suggested error handling will be refactore I'd ask for splliting of some methods especially those whos having several isolated if conditions and for cycle - especially translateDiscussion and importRepository functions.

Have you considered some unit tests for the key parts where some manipulation take place?

What about the SRP code isolation into the separate files based on logical context?

**Comment by klatys** [comment]
@marley-ma refactored to flatten the logic into separate functions and covered with tests where applicable.
As for the logging, to an extent I agree:
- I've added log of some previously ommited errors  in 6d598c77d1307befc212112400dba2861204c71e
- As for different approaches I don't think that error handling can/should be super unified:
  - where the error causes inability to run the script it's okay to raise the error and panic out (three places in this case - config read, both clients init)
  - where the error doesn't necessarily cause/call for a full break it's just logged and usually in the place where this happens - after all this is not SDK/Library where it actually makes sense to pass the errors upwards, here it would IMO harden the readability as you're just passing the errors higher and logging there anyway

**Review by mrlm-xyz — APPROVED** [review]

**Pull request merged**